specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,188 @@
+
+**nimble-angular**  
+One goal for the nimble-angular label support is to make it easy (or automatic) for clients to pick up new localized strings/labels when they uptake new nimble-angular versions. The way the prototype accomplishes that is:
+- Export (codegen) a [`NimbleLabels` class](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-8ada6f05dc0978bee3e574b5e06167631a98b8fd1ac8f5467aa3828e3ba933f1) that has properties for all Nimble labels, and uses the Angular `localize` function:
+```ts
+export class NimbleLabels {
+    public static readonly labelNumberFieldIncrement = $localize`Increment`;
+    public static readonly labelNumberFieldDecrement = $localize`Decrement`;
+}
+```
+(We may encode the label ID as the description so it shows up in the message files, i.e. ``$localize`:nimble-label-number-field-decrement:Decrement` ``).  
+If a consumer of nimble-angular imports the module containing the `NimbleLabels` class, once they run `ng extract-i18n` for their app, the message files will then contain the Nimble strings for translation. Additionally, once a client has done that, labels/strings in future nimble-angular version updates would automatically get pulled in by `ng extract-i18n` if it was run again after a package update.
+- Export (codegen) a directive which will automatically set all of the labels on a Nimble theme provider, with the values from `NimbleLabels`. In the prototype this is [`NimbleThemeProviderInitializeAllLabelsDirective`](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-a8bb0152269fed064bdc90ef28541bdedaeaf86f3a4e2b1751d2da68a6090aca) with selector `nimble-theme-provider[initializeAllLabels]`. Assuming that an app is OK with pulling in and localizing all of Nimble's strings/labels, using this directive will make the process of using those localized strings mostly automatic.
+- Codegen [`NimbleThemeProviderBaseDirective`](https://github.com/ni/nimble/commit/0b088a67af4a860fce17003e37ea1bf8dfd10e8b#diff-a339dcfbd672a173f9daf1b67e96ed59cd7de2167935aae338d76a277476b026): Similar to the nimble-components version, this exposes properties for all of the Nimble labels. This allows apps to pick & choose to only localize a few labels (e.g. if they're only using a subset of the controls), or override specific label values (set on an ancestor theme provider, or override the values from the *InitializeAllLabels directive). However the app wouldn't automatically pick up new strings if they only set labels this way.
+
+*Possible Alternatives:*
```

Comment:
Added a bullet point for that in Possible Alternatives. I think a directive per component is possible (though it would be hard to codegen like you said).